src/librustc_mir/interpret/memory.rs

```diff
@@ -834,7 +835,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
         src: impl IntoIterator<Item = u8>,
     ) -> InterpResult<'tcx> {
         let src = src.into_iter();
-        let size = Size::from_bytes(src.size_hint().0 as u64);
+        let size = Size::from_bytes(u64::try_from(src.size_hint().0).unwrap());
```

Comment:
Good thought, but I already took care of that in https://github.com/rust-lang/rust/pull/70226 so this will just cause merge conflict. Better to not do this change.